ion/textreader_test.go

```diff
@@ -423,6 +423,38 @@ func TestTrsToString(t *testing.T) {
 	}
 }
 
+func TestInStruct(t *testing.T) {
+	r := NewReaderStr("[ { a:() } ]")
+
+	r.Next()
+	r.StepIn() // In the list, before the struct
+	if r.InStruct() {
+		t.FailNow()
```

Comment:
`t.Fatal("InStruct returned true before we were in a struct")`?

Reply:
Done.